webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java

```diff
@@ -55,17 +65,23 @@ private void addListener() {
 
     userRegistrationLayout.registerButton.addClickListener(
         event -> {
-          resetErrorMessages();
+          clearNotifications();
           registrationUseCase.register(
               userRegistrationLayout.fullName.getValue(),
               userRegistrationLayout.email.getValue(),
               userRegistrationLayout.password.getValue().toCharArray());
         });
   }
 
-  private void resetErrorMessages() {
-    userRegistrationLayout.alreadyUsedEmailMessage.setVisible(false);
-    userRegistrationLayout.errorMessage.setVisible(false);
+  private void clearNotifications() {
+    userRegistrationLayout.notificationLayout.setVisible(false);
+    userRegistrationLayout.notificationLayout.removeAll();
+  }
+
+  private void showError(ErrorMessage errorMessage) {
+    userRegistrationLayout.notificationLayout.add(
+        new ErrorMessage(errorMessage.title(), errorMessage.message()));
+    userRegistrationLayout.notificationLayout.setVisible(true);
```

Comment:
same as above: either directly give the error message and not recreate it make the title and reason as parameters of the method and create the message there

Reply:
Good point 👍